src/IK/include/BipedalLocomotion/IK/InverseKinematics.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file InverseKinematics.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_IK_INVERSE_KINEMATICS_H
+#define BIPEDAL_LOCOMOTION_IK_INVERSE_KINEMATICS_H
+
+#include <memory>
+#include <string>
+#include <optional>
+
+#include <Eigen/Dense>
+#include <manif/SE3.h>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+#include <BipedalLocomotion/IK/Task.h>
+
+namespace BipedalLocomotion
+{
+
+namespace IK
+{
+
+/**
+ * State of the InverseKinematics
+ */
+struct IKState
+{
+    Eigen::VectorXd jointVelocity; /**< Joints velocity in rad per seconds */
+    manif::SE3d::Tangent baseVelocity; /**< Mixed spatial velocity of the base */
+};
+
+/**
+ * InverseKinematics implements the interface for the inverse kinematics. Please inherits this class
+ * if you want to implement your custom IK.
+ */
+class InverseKinematics : public BipedalLocomotion::System::Advanceable<IKState>
+{
+
+public:
+    /**
+     * Initialize the planner.
+     * @param handler pointer to the parameter handler.
+     * @return true in case of success/false otherwise.
+     */
+    virtual bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> handler);
+
+    /**
+     * Add a task in the inverse kinematics
+     * @param task pointer to a given task
+     * @param priority is the priority of the task. The lower the number the higher the priority.
+     * @param weight weight associated to the task.
+     * @param isInequality true if the task should be treated as inequality constraint
+     * @return true if the task has been added to the inverse kinematics.
+     */
+    virtual bool addTask(std::shared_ptr<Task> task,
+                         const std::string& taskName,
+                         std::size_t priority,
+                         std::optional<Eigen::Ref<const Eigen::VectorXd>> weight = {},
+                         bool isInequality = false) = 0;
+
+    /**
+     * Get a vector containing the name of the tasks.
+     * @return an std::vector containing all the names associated to the tasks
+     */
+    virtual std::vector<std::string> getTasksName() const = 0;
```

Comment:
Would it be more grammatically correct ``getTaskNames``?

Reply:
Done in b7a79cab9bac0e934e9807bbdfa2e00a098f7574 and f05ee3afdcaa596e27f258a17ca05827b1d03a6c